pipeline/backend/kubernetes/kubernetes.go

```diff
@@ -1,28 +1,54 @@
 package kubernetes
 
 import (
+	"bytes"
 	"context"
+	"fmt"
 	"io"
 	"os"
+	"strings"
+	"time"
 
 	"github.com/woodpecker-ci/woodpecker/pipeline/backend/types"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+
+	// To authenticate to GCP K8s clusters
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
 )
 
 type kube struct {
-	namespace string
-	endpoint  string
-	token     string
+	logs         *bytes.Buffer // TODO remove
+	namespace    string
+	storageClass string
+	volumeSize   string
+	client       kubernetes.Interface
 }
 
-// make sure kube implements Engine
-var _ types.Engine = &kube{}
-
 // New returns a new Kubernetes Engine.
-func New(namespace, endpoint, token string) types.Engine {
+func New() types.Engine {
+	namespace := os.Getenv("WOODPECKER_BACKEND_K8S_NAMESPACE")
+	if namespace == "" {
+		namespace = "woodpecker-ci"
+	}
+	storageClass := os.Getenv("WOODPECKER_BACKEND_K8S_NAMESPACE")
+	volumeSize := os.Getenv("WOODPECKER_BACKEND_K8S_NAMESPACE")
```

Comment:
This causes a panic:

```
Defaulted container "woodpecker-agent" out of: woodpecker-agent, dind
{"level":"error","Repo":"stintel/vallumd","Build":"5","ID":"75","error":"pods \"0-7304064393753199929-clone\" not found","time":"2022-08-14T21:17:27Z","message":"could not destroy engine"}
panic: cannot parse 'woodpecker-agents': quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'

goroutine 15 [running]:
k8s.io/apimachinery/pkg/api/resource.MustParse({0xc000044021, 0x11})
        /src/vendor/k8s.io/apimachinery/pkg/api/resource/quantity.go:128 +0x1ce
github.com/woodpecker-ci/woodpecker/pipeline/backend/kubernetes.PersistentVolumeClaim({0xc000044021, 0x11}, {0xc00050c380, 0x1d}, {0xc000044021, 0x11}, {0xc000044021, 0x11})
        /src/pipeline/backend/kubernetes/volume.go:27 +0xf2
github.com/woodpecker-ci/woodpecker/pipeline/backend/kubernetes.(*kube).Setup(0xc0001199f0, {0x1ae9878, 0xc000216900}, 0xc000214600)
        /src/pipeline/backend/kubernetes/kubernetes.go:87 +0xf4
github.com/woodpecker-ci/woodpecker/pipeline.(*Runtime).Run(0xc000226690)
        /src/pipeline/pipeline.go:93 +0x2f5
github.com/woodpecker-ci/woodpecker/agent.(*Runner).Run(0xc0000b1b80, {0x1ae9878, 0xc0000b0f80})
        /src/agent/runner.go:319 +0xea5
main.loop.func3()
        /src/cmd/agent/agent.go:169 +0x432
created by main.loop
        /src/cmd/agent/agent.go:142 +0xe99
```

Reply:
To elaborate, both storageClass and volumeSize use the same env var WOODPECKER_BACKEND_K8S_NAMESPACE

Reply:
I worked around it by naming my namespace 10240m, but then I got stuck because I don't have a ReadWriteMany StorageClass in my poor man's K3S cluster.

Reply:
New version is currently building with support for `WOODPECKER_BACKEND_K8S_NAMESPACE`, `WOODPECKER_BACKEND_K8S_VOLUME_SIZE`, `WOODPECKER_BACKEND_K8S_STORAGE_CLASS`: https://ci.woodpecker-ci.org/woodpecker-ci/woodpecker/build/3160

Reply:
Meanwhile I managed to get things to work using https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner. Still a poor man's K3S cluster, but it's working nicely. I'd say with those copy-paste errors fixed and some decent documentation, this is good to go. Thanks!

Reply:
New agent version got pushed.